TRRandomizerCore/Randomizers/TR1/TR1ItemRandomizer.cs

```diff
@@ -356,25 +397,83 @@ private void RandomizeKeyItems(TR1CombinedLevel level)
 
         _picker.Initialise(_levelInstance.Name, GetItemLocationPool(_levelInstance, true), Settings, _generator);
 
+        if (level.Is(TR1LevelNames.TIHOCAN))
+        {
+            // Enemy rando may not be selected or Pierre may have ended up at the
+            // end as usual. Remove his key and scion drops and place them as items.
+            if (level.Data.Entities[82].TypeID == TR1Type.Pierre)
```

Comment:
Hardcoded 82

Reply:
Thanks, made it a const. We never change enemy indices so it's safe enough, but definitely better to make it fixed.